lib/db/set.go

```diff
@@ -380,6 +381,44 @@ func (s *Snapshot) RemoteNeedFolderFiles(device protocol.DeviceID, page, perpage
 	return files
 }
 
+func (s *Snapshot) WithBlocksHash(hash []byte, iterator Iterator) error {
+	key, err := s.t.keyer.GenerateBlockListMapKey(nil, []byte(s.folder), hash, nil)
+	if err != nil {
+		return err
+	}
+
+	iter, err := s.t.NewPrefixIterator(key)
+	if err != nil {
+		return err
+	}
+	defer iter.Release()
+
+	for iter.Next() {
+		file := string(s.t.keyer.NameFromBlockListMapKey(iter.Key()))
+		fi, ok := s.Get(protocol.LocalDeviceID, file)
+		if !ok {
+			continue
+		}
+
+		if !bytes.Equal(fi.BlocksHash, hash) {
+			// Perhaps should be at least info?
```

Comment:
I'd say either we are confident this doesn't happen, then leave it at debug (or don't even check), or make it a warning to get reports asap (and maybe even create a repair mechanism like with sequences, but 🤮).